images/recycler/scripts/recycler.sh

```diff
@@ -22,8 +21,10 @@ if [[ ! -d ${dir} ]]; then
     exit 1
 fi
 
-# shred all files
-find ${dir} -type f -exec shred -fuvz {} \;
+find ${dir} -type f -exec stat -c "%u %g %n"  {} \; |xargs -n 3 bash -c 'sudo -u "#$1" shred  $3' argv0
```

Comment:
inspired by @liggitt 
the following shred files, rm non-directories, and rm directories in a row

```bash
# shred regular files
function recycle_file() {
    filename="${1}"
    uid=$(stat -c "#%u" "${filename}")
    gid=$(stat -c "#%g" "${filename}")
    sudo -u "${uid}" shred "${filename}"
}
export -f recycle_file

find ${tmp} -type f -print0 | xargs -n 1 -0 bash -c 'recycle_file "$@"' {}

# rm others
function rm_all() {
    filename="${1}"
    uid=$(stat -c "#%u" "${filename}")
    gid=$(stat -c "#%g" "${filename}")
    sudo -u "${uid}" rm -rf "${filename}"
}
export -f rm_all

find ${tmp} ! -type d -print0 | xargs -n 1 -0 bash -c 'rm_all "$@"' {}


#find ${dir} -type f -exec stat -c "%u %g %n"  {} \; |xargs -n 3 bash -c 'sudo -u "#$1" shred  $3' argv0

#delete directories
find ${tmp} -mindepth 1  -type d  -exec stat -c "%n %u %g"  {} \;|sort -k 1 -rg | xargs -n 3 bash -c 'sudo -u "#$2" rm -rf $1' argv0
```